logic/transactions/pool.js

```diff
@@ -0,0 +1,922 @@
+	}
+
+	jobsQueue.register('transactionPoolNextReset', nextReset, self.poolExpiryInterval);
+
+	if (cbPool) {
+		return setImmediate(cbPool, null, self);
+	}
+}
+
+
+// Private
+/**
+ * Gets all or limited number of transactions from input object list and returns ordered array.
+ * @private
+ * @param {transaction{}} transactions
+ * @param {boolean} reverse
+ * @param {number} limit
+ * @return {transaction[]}
+ */
+__private.getTransactionsFromPoolList = function (transactionsInPool, reverse, limit) {
+	var transactions = _.orderBy(transactionsInPool, ['receivedAt'],['asc']);
+
+	transactions = reverse ? transactions.reverse() : transactions;
+
+	if (limit) {
+		transactions.splice(limit);
+	}
+
+	return transactions;
+};
+
+/**
+ * Gets transactions from all pool lists based on filter.
+ * @private
+ * @param {Object} filter search criteria
+ * @return {Objetc} transactions by pool list
+ */
+__private.getAllPoolTransactionsByFilter = function (filter) {
+	var transactions = {
+		unverified: _.filter(pool.unverified.transactions, filter),
+		pending: _.filter(pool.verified.pending.transactions, filter),
+		ready: _.filter(pool.verified.ready.transactions, filter)
+	};
+
+	return transactions;
+};
+
+/**
+ * Returns true if the id is present in at least one of the pool lists transactions.
+ * @private
+ * @param {string} id
+ * @return {boolean}
+ */
+__private.transactionInPool = function (id) {
+	return [
+		pool.verified.ready.transactions[id],
+		pool.unverified.transactions[id],
+		pool.verified.pending.transactions[id]
+	].some(function (index) {
+		return typeof(index) === 'object';
+	});
+};
+
+/**
+ * Checks if pool is ready to receive a new transaction and the incoming
+ * transaction was not already processed by pool.
+ * @private
+ * @implements {__private.countTransactionsPool}
+ * @implements {__private.transactionInPool}
+ * @param {Object} transaction
+ * @param {function} cb - Callback function.
+ * @return {setImmediateCallback} error | cb, transaction
+ */
+__private.checkPoolAvailability = function (transaction, waterCb) {
+	if (__private.countTransactionsPool() >= self.poolStorageTransactionsLimit) {
+		return setImmediate(waterCb, 'Transaction pool is full');
+	}
+	if (pool.invalid.transactions[transaction.id] !== undefined) {
+		return setImmediate(waterCb, 'Transaction is already processed as invalid: ' + transaction.id);
+	}
+	if (__private.transactionInPool(transaction.id)) {
+		return setImmediate(waterCb, 'Transaction is already in pool: ' + transaction.id);
+	}
+	return setImmediate(waterCb, null, transaction);
+};
+
+/**
+ * Obtains sender from accounts table, if address does not exists,
+ * returns a default account object.
+ * @private
+ * @implements {modules.accounts.setAccountAndGet}
+ * @param {Object} transaction
+ * @param {function} cb - Callback function.
+ * @return {setImmediateCallback} error | cb, transaction, sender
+ */
+__private.setAccountAndGet = function (transaction, waterCb) {
```

Comment:
it will check if account exists in database.
Now: if doesn't exists, create it
Future: for type:0, if doesn't  exists return an object with a temporal account